### PR TITLE
automatically squash sti-base

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@soltysh ptal.

**Comment by bparees** [comment]
@goldmann thanks for the awesome squashing tool.

**Comment by goldmann** [comment]
You welcome! If you find any issues or have RFEs - please [file a ticket](https://github.com/goldmann/docker-scripts/issues).

File: hack/build.sh [hunk 1]
```diff
@@ -5,6 +5,23 @@ OS=$1
 
 IMAGE_NAME=openshift/base-${OS}
 
+function squash {	
+	# install the docker layer squashing tool
+	rm -rf docker-scripts
+	git clone https://github.com/goldmann/docker-scripts/
```
Comment (goldmann) [inline hack/build.sh]: Please make sure you use a tag. Probably I'll introduce some major changes to the structure of the repo.

File: hack/build.sh [hunk 2]
```diff
@@ -5,6 +5,23 @@ OS=$1
...
+	pip=$(which pip)
+	if [ ! -x $pip ]; then 
+		wget https://bootstrap.pypa.io/get-pip.py
```
Comment (soltysh) [inline hack/build.sh]: You could use `easy_install`, like we do in sti-python, see [here](https://github.com/openshift/sti-python/blob/master/3.3/Dockerfile#L30).

**Comment by soltysh** [comment]
Awesome job @bparees, thx @goldmann for the tool :+1: 
LGTM

File: hack/build.sh (hunk 1 above)
Comment (goldmann) [inline hack/build.sh]: BTW, I work on publishing it to PyPi, but this will affect how you call the tool. I'll let you know once it's finished.

**Comment by mfojtik** [comment]
LGTM (@goldmann, how about rewriting the tool to Go? :-)

**Comment by goldmann** [comment]
@mfojtik Haha! No. :)

**Comment by mfojtik** [comment]
@goldmann (sad) :panda_face:

**Comment by mfojtik** [comment]
@goldmann that will allow to distribute it as a single binary without dependencies and it will allow us to use it in STI or other tooling :-)

**Comment by goldmann** [comment]
Having it in Python makes it easy to integrate with rel-eng tools.

**Comment by soltysh** [comment]
@mfojtik with @goldmann's approval we could include it as part of STI in future versions.

**Comment by mfojtik** [comment]
@soltysh if we rewrite this to Go :-)

**Comment by mfojtik** [comment]
@soltysh it will be cool feature of STI to squash the base image and then just add the application layer. So you can have just base and application layers.

**Comment by goldmann** [comment]
@mfojtik Do not touch base layers, this will create a lot of unnecessary data to move around since squashed layers will have new IDs. Squash only what you add (application layer).

**Comment by mfojtik** [comment]
@goldmann in that case squashing in STI does not make sense, because we only add the application layer which is just one layer (I believe).

**Comment by soltysh** [comment]
@mfojtik if so, we should apply the squashing just to our builder images, like @bparees is doing that with the base one.

**Comment by bparees** [comment]
@soltysh we discussed that we don't want to squash our builder images because we don't want minor changes to the assemble/run scripts to force a full re-pull of the entire builder layer.

however now that i restructured the builder layers to put the ADD early in the dockerfile (in order to reduce the number of layers) we may not have that benefit any more anyway.  But we do only introduce 3-4 layers in our builders now.

**Comment by soltysh** [comment]
@mfojtik btw, just found that https://github.com/jwilder/docker-squash

File: hack/build.sh (hunk 2 above)
Comment (bparees) [inline hack/build.sh]: thanks, will do.

**Comment by bparees** [comment]
@soltysh i only looked at it briefly but, that tool doesn't seem to support merging an arbitrary subset of layers like @goldmann 's tool does, which means we couldn't keep centos as its own set of layers.

**Pull request merged**